view.js

```diff
@@ -21,44 +21,105 @@ $('#add-to-list').on('click', () => {
 })
 
 
+
+$('#add-to-list-from-csv').on('click', () => {
+
+ var fileUpload = document.getElementById("fileUpload");
+
+    if (!fileUpload.value.endsWith(".csv") && !fileUpload.value.endsWith(".txt")){
+        alert("Please upload a valid CSV file.");
+    } else if (typeof(FileReader) == "undefined" && typeof(Storage) == "undefined"){
+        alert("This browser does not support HTML5.");
+    } else {
+        do_upload(fileUpload);
+
+    }
+})
+
+
+function do_upload(file){
+    var reader = new FileReader();
+    reader.readAsText(fileUpload.files[0]);
+    
+    reader.onload = (e) => {
+        var rows = e.target.result.split("\r\n");
+        var groups = {};
+
+        if (rows.length < 1) {
+            alert("Empty csv file");
+            return
+        }
+
+        for (var i = 0; i < rows.length; i++) {
+            if(rows[i] == "") continue
+         
+            const cells = rows[i].split(",");
+          if(cells[0] == "") continue
+            const group_number = cells[3];
+            const student_detail = {name: cells[0], gh_handle: cells[1], repository: cells[2]};
+
+            if (group_number in groups){
+                groups[group_number].push(student_detail)
+            } else {
+                groups[group_number] = [student_detail];
+            }
+        }
+
+        for (var key in groups) {
+            localStorage.setItem(key, JSON.stringify(groups[key]));
+            email=JSON.stringify(groups[key]);
+            addEntry(key, email)
+        }
+    }
+    document.getElementById("fileUpload").value = "";
+}
+
+
+
 $('#del-from-list').on('click', () => {
-	fs.unlink(filename, function(err)
-	{
-		if(err)
-		{
-			return console.error(err)
-		}
-		console.log("file deleted")
-		sno=0
-		$('#contact-table td').remove()
-	})
+	// fs.unlink(filename, function(err)
+	// {
+	// 	if(err)
+	// 	{
+	// 		return console.error(err)
+	// 	}
+	// 	console.log("file deleted")
+	// 	sno=0
+	// 	$('#contact-table td').remove()
+	// })
+  console.log("file deleted")
+    sno=0
```

Comment:
Where is this used?

Reply:
I guess sno is used while printing the table in the popup.

Reply:
Is `sno` global?

Reply:
Yes

Reply:
😱